wavedump/src/analyze-waveforms

```diff
@@ -55,6 +26,26 @@ def get_times(x, data, baseline=10):
     threshold = 0.4*data[np.arange(data.shape[0]),argmin]
     return x[data.shape[1]-np.argmax(((np.arange(data.shape[1]) < argmin[:,np.newaxis]) & (data > threshold[:,np.newaxis]))[:,::-1],axis=-1)-1]
 
+def get_spe_window(x, start, integration_time):
+    """
+    Returns the indicies over which the SPE analysis should be integrated.
+
+    `start` and `integration_time` are in nanoseconds, not indexes.
+    `start` is relative to the trigger, so `start = 0` is not 
+    the first sample, but rather the time the trigger fired.
+    """
+    time_per_index = x[1] - x[0]
+    a = int(np.abs(x - start).argmin())
+    if a >= len(x) - 1:
+        print('SPE integration start is at max. Setting it to ten below max.')
+        a = len(x) - 11
+    b = int(np.round(a + integration_time / time_per_index))
+    if b > len(x) - 1:
+        print('SPE integration time is too long. Setting upper bound to max.')
+        b = len(x) - 1
```

Comment:
This could potentially be useful, but I think it's still probably better to just quit.